src/main/java/io/aiven/kafka/connect/common/output/jsonwriter/JsonLinesOutputWriter.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2020 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.common.output.jsonwriter;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Objects;
+
+import org.apache.kafka.connect.errors.ConnectException;
+import org.apache.kafka.connect.errors.DataException;
+import org.apache.kafka.connect.sink.SinkRecord;
+
+import io.aiven.kafka.connect.common.config.OutputField;
+import io.aiven.kafka.connect.common.config.OutputFieldType;
+import io.aiven.kafka.connect.common.output.OutputWriter;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+
+public class JsonLinesOutputWriter implements OutputWriter {
+
+    private final Map<String, OutputFieldBuilder> writers;
+    private static final byte[] RECORD_SEPARATOR = "\n".getBytes(StandardCharsets.UTF_8);
+    private final ObjectMapper objectMapper;
+
+    JsonLinesOutputWriter(final Map<String, OutputFieldBuilder> writers) {
+
+        this.writers = writers;
+        this.objectMapper = new ObjectMapper();
+        objectMapper.setNodeFactory(JsonNodeFactory.withExactBigDecimals(true));
+    }
+
+    public void writeRecord(final SinkRecord record,
+                            final OutputStream outputStream) throws IOException {
+        Objects.requireNonNull(record, "record cannot be null");
+        Objects.requireNonNull(outputStream, "outputStream cannot be null");
+
+        try {
+            outputStream.write(objectMapper.writeValueAsBytes(writeFields(record)));
+            outputStream.write(RECORD_SEPARATOR);
+        } catch (final Exception e) {
+            throw new DataException("Converting Kafka Connect data to byte[] failed due to serialization error: ", e);
```

Comment:
Practically, because we are using only JsonConvertor, JsonProcessingException would never be an option. I should always return only DataException. BUT! Because it is in `readTree` description,  we cannot narrow down interface to a DataException.